storageapi/src/vespa/storageapi/mbusprot/storageprotocol.cpp

```diff
@@ -110,10 +119,13 @@ StorageProtocol::encode(const vespalib::Version& version,
         }
 
     } catch (std::exception & e) {
-        LOGBP(warning, "Failed to encode %s storage protocol message %s: %s",
-              version.toString().c_str(),
-              message.getInternalMessage()->toString().c_str(),
-              e.what());
+        if (version < version6_0 &&
+            !suppressEncodeWarning(message.getInternalMessage().get())) {
```

Comment:
With this we cannot get any warnings if other encodes fail on >= `version6_0`. Do we instead want
```c++
if (!(version < version6_0 && suppressEncodeWarning(message.getInternalMessage().get()))) {
```
?